src/Core/Services/StateService.cs

```diff
@@ -46,6 +46,19 @@ public async Task<string> GetActiveUserIdAsync()
             return activeUserId;
         }
 
+        public async Task<string> GetActiveUserEmailAsync()
+        {
+            await CheckStateAsync();
+
+            var activeUserId = _state?.ActiveUserId;
+            if (activeUserId == null)
+            {
+                var state = await GetStateFromStorageAsync();
+                activeUserId = state?.ActiveUserId;
+            }
+            return await GetEmailAsync(activeUserId);
+        }
```

Comment:
Don't duplicate code. Either call from the caller:
```c#
var activeUserId = await _stateService.GetActiveUserIdAsync();
var activeUserEmail = await _stateService.GetEmailAsync(activeUserId);
```
or here just do:
```c#
public async Task<string> GetActiveUserEmailAsync()
{
    var activeUserId = await GetActiveUserIdAsync();
    return await GetEmailAsync(activeUserId);
}
```

Reply:
my bad...